tests/test_authentication.py

```diff
@@ -66,3 +75,225 @@ async def test_authentication(fastapi_app, client, monkeypatch, httpx_mock: HTTP
+    """
+    Test that the http endpoints for the adgs station are protected and return 401 if not authorized.
+    Set RSPY_LOCAL_MODE to False before running the fastapi app.
+    """
+
+    # Mock the uac manager url
+    monkeypatch.setenv("RSPY_UAC_CHECK_URL", RSPY_UAC_CHECK_URL)
+
+    read_type = "read"
+    download_type = "download"
+    read_request_params = {"datetime": "2014-01-01T12:00:00Z/2023-02-02T23:59:59Z"}
+    download_request_params = {"name": "TEST_FILE.raw"}
+    endpoints = []
+    # Gather the endpoints
+    # For each api endpoint (except the technical endpoints)
+    for route in fastapi_app.router.routes:
+        if (not isinstance(route, APIRoute)) or (route.path in ("/", "/health")) or "adgs" not in route.path:
+            continue
+
+        # For each method (get, post, ...)
+        for method in route.methods:
+            logger.debug(f"\n\nTest the {route.path} [{method}] authentication")
+
+            if "search" in route.path:
+                endpoints.append((read_type, route.path + "?" + urllib.parse.urlencode(read_request_params), method))
+                # this means the authorization shall pass
+            else:
+                endpoints.append(
+                    (download_type, route.path + "?" + urllib.parse.urlencode(download_request_params), method),
+                )
+
+    # With a valid api key in headers, set the read role
+    # Clear the cache
+    ttl_cache.clear()
+    httpx_mock.add_response(
+        url=RSPY_UAC_CHECK_URL,
+        match_headers={APIKEY_HEADER: VALID_APIKEY},
+        status_code=HTTP_200_OK,
+        json={"iam_roles": ["rs_adgs_read"], "config": {}},
+    )
+
+    for endpoint in endpoints:
+        if endpoint[0] == download_type:
+            expected_response = [HTTP_401_UNAUTHORIZED]
+        else:
+            # this means the authorization shall pass
+            expected_response = [HTTP_503_SERVICE_UNAVAILABLE, HTTP_404_NOT_FOUND]
+        assert (
+            client.request(endpoint[2], endpoint[1], headers={APIKEY_HEADER: VALID_APIKEY}).status_code
+            in expected_response
+        )
+
+    # With a valid api key in headers, set the download role
+    # Clear the cache
+    ttl_cache.clear()
+    httpx_mock.add_response(
+        url=RSPY_UAC_CHECK_URL,
+        match_headers={APIKEY_HEADER: VALID_APIKEY},
+        status_code=HTTP_200_OK,
+        json={"iam_roles": ["rs_adgs_download"], "config": {}},
+    )
+
+    for endpoint in endpoints:
+        if endpoint[0] == read_type:
+            expected_response = [HTTP_401_UNAUTHORIZED]
+        else:
+            # this means the authorization shall pass
+            expected_response = [HTTP_503_SERVICE_UNAVAILABLE, HTTP_404_NOT_FOUND]
+
+        assert (
+            client.request(endpoint[2], endpoint[1], headers={APIKEY_HEADER: VALID_APIKEY}).status_code
+            in expected_response
+        )
+
```

Comment:
This whole block can be replaced with: 

`
    assert all(
        client.request(method, url, headers={APIKEY_HEADER: VALID_APIKEY}).status_code in
        ([HTTP_401_UNAUTHORIZED] if request_type == download_type else [HTTP_503_SERVICE_UNAVAILABLE,
                                                                        HTTP_404_NOT_FOUND])
        for request_type, url, method in endpoints
    )
`